paddle/framework/channel_test.cc

```diff
@@ -60,6 +60,32 @@ TEST(Channel, SufficientBufferSizeDoesntBlock) {
   delete ch;
 }
 
+// This test tests that CloseChannel returns a value of zero
+// immediately to all receivers that are trying to receive from the channel.
+TEST(Channel, ReceiverGetsZeroOnClosedChannel) {
+  const size_t buffer_size = 10;
+  auto ch = MakeChannel<size_t>(buffer_size);
```

Comment:
This invariant should be verified with buffered and unbuffered channels.

Reply:
On a side note, correct me if I am wrong @wangkuiyi but for an unbuffered channel, since the buffer size is zero, there would be no residual values. So trying to read from a closed channel should just return a zero value, right ?